Minor spec fix for ADL job types

begoldsm opened the pull request:
Change offset to be an int instead of string
Add in enum values to type.

Commit:
Parameters are not required for adding a datalake store

Updating the spec.

Commit:
this should be an integer, not a string.

Commit:
Reverting required change

This is due to bug: 7075103

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
make type an explicit enum.

amarzavery commented on arm-datalake-analytics/job/2015-11-01-preview/swagger/job.json:
```diff
@@ -706,6 +706,10 @@
+          "enum": [
+            "USql",
+            "Hive"
+          ],
```
Did you have a look at the x-ms-enum extension doc over here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#Enum-x-ms-enum

Commit:
Add x-ms-enum per Amar's comment

This value will rarely, if ever, change.

Commit:
Removing because this is part of a discriminator

enum types aren't used directly for discriminators. Instead it is
defined by each valid discriminator type.
